src/pelicanfs/core.py

```diff
@@ -256,7 +311,15 @@ async def _expand_path(self, path, recursive=False, maxdepth=None):
 fsspec.register_implementation(PelicanFileSystem.protocol, PelicanFileSystem)
 
 def PelicanMap(root, pelfs, check=False, create=False):
+    if pelfs.specifiedCache:
+        return pelfs.get_mapper(pelfs.specifiedCache + "/" + root, check=check, create=create)
+
+
     loop = asyncio.get_event_loop()
-    cache_url = loop.run_until_complete(pelfs.get_working_cache(root))
+    if pelfs.directReads:
+        dataUrl = loop.run_until_complete(pelfs.get_origin_url(root))
+
+    else:
+        dataUrl = loop.run_until_complete(pelfs.get_working_cache(root))
 
-    return pelfs.get_mapper(cache_url, check=check, create=create)
\ No newline at end of file
+    return pelfs.get_mapper(dataUrl, check=check, create=create)
```

Comment:
Check your IDE settings -- it should be adding a newline at the end of the file here.